### PR TITLE
OKTA-600908 next to master

### PR DESCRIPTION
**Pull request by glenfannin-okta**
## Description:
Merging SIW Next feature branch into SIW Master.

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-600908](https://oktainc.atlassian.net/browse/OKTA-600908)

### Downstream Monolith Build:

https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-0a39763-6480ad70&page=1&pageSize=6&sha=67ce31621d4babe693623aa3d3ac44207cc56933&tab=main

### CODE REVIEW

File: scripts/verify-package.sh [hunk 1]
```diff
@@ -119,6 +119,12 @@ fi
 
 mkdir -p "test-reports/verify-package"
 
+# Must switch the node version back to v14.18.2 due to bug in v16 see https://github.com/npm/cli/pull/5894
+if ! setup_service node v14.18.2 &> /dev/null; then
+  echo "Failed to install node"
+  exit ${FAILED_SETUP}
+fi
```
Comment (glenfannin-okta) [inline scripts/verify-package.sh:126]: There was a bug in npm that caused `npm pack` to produce a pack report with a filename that contained the package scope instead of hyphens. Our verify package test validated against this file name and failed when using node v16.19.1, we have to explicitly set it back to v14.18.2 to get it to work.

File: scripts/buildtools/commands/verify-package.js [hunk 2]
```diff
@@ -3,8 +3,8 @@ const { readFileSync } = require('fs');
 
 const KB = 1024;
 const MB = 1024 * 1024;
-const EXPECTED_PACKAGE_SIZE = 13.35 * MB;
-const EXPECTED_PACKAGE_FILES = 2200;
+const EXPECTED_PACKAGE_SIZE = 45 * MB;
+const EXPECTED_PACKAGE_FILES = 21000;
```
Comment (glenfannin-okta) [inline scripts/buildtools/commands/verify-package.js:7]: I had to increase these package and file size/counts because of the new results generated by `npm pack` with Gen 3 bundled. Looking to see get some eyes on this to determine if this is acceptable?

Comment (shawyu-okta) [reply scripts/buildtools/commands/verify-package.js:7]: I think these numbers are the total size of _all_ of the bundle outputs? If that's the case then I would expect both of these to increase

File: scripts/buildtools/commands/verify-package.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ const EXPECTED_BUNDLE_SIZES = {
   'okta-sign-in.oie.min.js': 1.2 * MB,
   'okta-sign-in.polyfill.js': 504 * KB,
   'okta-sign-in.polyfill.min.js': 108 * KB,
-  'okta-sign-in.next.js': 7 * MB, // NOTE: bloated temporarily without tree-shaking
+  'okta-sign-in.next.js': 1.6 * MB, // NOTE: bloated temporarily without tree-shaking
```
Comment (shawyu-okta) [inline scripts/buildtools/commands/verify-package.js:20]: Oh, much smaller now! What happened?

File: src/v3/widgetrc.ts [hunk 4]
```diff
@@ -68,7 +68,7 @@ const common: Partial<WidgetOptions> = {
     // autoFocus: true,
     // hideSignOutLinkInMFA: true,
     // mfaOnlyFlow: true,
-    rememberMe: false,
+    rememberMe: true,
```
Comment (shawyu-okta) [inline src/v3/widgetrc.ts:71]: Just checking this is the "correct" default state

File: src/v3/vite.config.ts [hunk 5]
```diff
@@ -95,8 +95,8 @@ export default defineConfig(({ mode, command }) => {
       return {
         ...base,
 
-        // hide sourcemaps
-        sourcemap: false, // boolean | 'inline' | 'hidden'
+        // generate sourcemaps
+        sourcemap: true, // boolean | 'inline' | 'hidden'
```
Comment (shawyu-okta) [inline src/v3/vite.config.ts:99]: Is this change intentional?

File: assets/sass/modules/_app-login-banner.scss [hunk 6]
```diff
@@ -41,12 +41,6 @@
       font-weight: lighter;
       line-height: 26px;
     }
-
-    p {
-      @include max-height-mq(660px) {
-        display: none;
-      }
-    }
```
[the remark this answers is not in the document]
Comment (shawyu-okta) [reply assets/sass/modules/_app-login-banner.scss:49]: Yeah, we can follow up for G3 if we're not sure

File: packages/@okta/courage-for-signin-widget/yarn.lock [hunk 7]
```diff
@@ -584,9 +584,9 @@
     "@types/underscore" "*"
 
 "@types/eslint-scope@^3.7.0":
-  version "3.7.4"
-  resolved "https://artifacts.aue1e.internal:443/artifactory/api/npm/npm-okta-master/@types/eslint-scope/-/eslint-scope-3.7.4.tgz#37fc1223f0786c39627068a12e94d6e6fc61de16"
-  integrity sha1-N/wSI/B4bDlicGihLpTW5vxh3hY=
+  version "3.7.1"
```
Comment (shawyu-okta) [inline packages/@okta/courage-for-signin-widget/yarn.lock:587]: This version actually decreased, is that right?

**Review by shawyu-okta — APPROVED** [review]

File: scripts/buildtools/commands/verify-package.js [hunk 8]
```diff
@@ -15,8 +15,8 @@ const EXPECTED_BUNDLE_SIZES = {
   'okta-sign-in.no-polyfill.min.js': 1.6 * MB,
   'okta-sign-in.oie.js': 2.4 * MB,
   'okta-sign-in.oie.min.js': 1.2 * MB,
-  'okta-sign-in.polyfill.js': 504 * KB,
-  'okta-sign-in.polyfill.min.js': 108 * KB,
+  'okta-sign-in.polyfill.js': 380 * KB,
+  'okta-sign-in.polyfill.min.js': 90 * KB,
```
[the remark this answers is not in the document]
Comment (glenfannin-okta) [reply scripts/buildtools/commands/verify-package.js:19]: Yea I am not sure _why_ it decreased in size as we didnt really touch anything in Gen 2, initially I thought vite had something to do with it but we are only using vite for Gen 3 so that can't be it 🤔

File: scripts/buildtools/commands/verify-package.js (hunk 3 above)
Comment (glenfannin-okta) [reply scripts/buildtools/commands/verify-package.js:20]: Oh yea, this _has_ to be because of the addition of vite, I recall when Lester added vite and used it to build it decreased the js file size down drastically. Since adding vite, we never came back and updated this verify package script so this is the first time its being reflected 🙏

File: src/v3/widgetrc.ts (hunk 4 above)
Comment (glenfannin-okta) [reply src/v3/widgetrc.ts:71]: Yessir! In [v2](https://github.com/okta/okta-signin-widget/blob/master/src/models/Settings.ts#L84) it is default to true, so I updated this to reflect that in Gen3

File: src/v3/vite.config.ts (hunk 5 above)
Comment (glenfannin-okta) [reply src/v3/vite.config.ts:99]: Yessir! The verify-package.js file expects the `okta-signin-widget.next.js` file to have a corresponding `okta-signin-widget.next.js.map` file for the sourcemaps, and if it does not exist it will fail the assertion. This setting generates that sourcemap file for us.

File: packages/@okta/courage-for-signin-widget/yarn.lock (hunk 7 above)
Comment (glenfannin-okta) [reply packages/@okta/courage-for-signin-widget/yarn.lock:587]: Yep, it looks like in the Gen3 branch we allowed the patch update but when I compared it to Gen2 (master branch), it was set to this earlier version so I accepted this change instead of upgrading to prevent issues.
See: https://github.com/okta/okta-signin-widget/blob/master/packages/%40okta/courage-for-signin-widget/yarn.lock#L586-L592

**Review by shawyu-okta — APPROVED** [review]

**Comment by shawyu-okta** [comment]
re-approved, i reviewed your responses to comments :)

File: packages/@okta/i18n/src/properties/login.properties [hunk 9]
```diff
@@ -1493,4 +1498,25 @@ idx.error.user.not.assigned.to.app = User is not assigned to this application.
 idx.error.code.user_not_assigned = You are not allowed to access this app. To request access, contact an admin.
 
 # Custom Identifiers
-custom.identifiers.userNotResolved = Can't identify your account with {0}. Enter your username.
\ No newline at end of file
+custom.identifiers.userNotResolved = Can't identify your account with {0}. Enter your username.
```
Comment (zihanwang-okta) [inline packages/@okta/i18n/src/properties/login.properties:1501]: For all the argument `{0}`, could you please add a short description. It will give our translator/vendor some more context, so that we can have a better translation quality. Example: line 877

File: scripts/buildtools/commands/verify-package.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (lesterchoi-okta) [reply scripts/buildtools/commands/verify-package.js:19]: Gonna dig into this. Smaller sounds better but not if it's because we're not missing polyfills that should be present.

[the remark this answers is not in the document]
Comment (lesterchoi-okta) [reply scripts/buildtools/commands/verify-package.js:19]: 1. Did our version of `core-js` change? 
2. Did we change our targets? 
3. Did we change which/how polyfills are included?

All the changes I see are related to path changes and polyfills used.

Big chunk omitted from topic branch side:
https://gist.github.com/lesterchoi-okta/bf1d431ed809decd415daf5f45f26dfa#file-okta-sign-in-polyfill-js-diff-L1-L183 

Example path change:
https://gist.github.com/lesterchoi-okta/bf1d431ed809decd415daf5f45f26dfa#file-okta-sign-in-polyfill-js-diff-L184-L187

Example bundling change:
https://gist.github.com/lesterchoi-okta/bf1d431ed809decd415daf5f45f26dfa#file-okta-sign-in-polyfill-js-diff-L217-L222

File: playground/mocks/data/idp/idx/enroll-profile-new-custom-labels.json [hunk 10]
```diff
@@ -21,17 +21,20 @@
                 {
                   "name": "lastName",
                   "label": "Please enter your last name",
-                  "required": true
+                  "required": true,
+                  "customLabel": true
```
[the remark this answers is not in the document]
Comment (glenfannin-okta) [reply playground/mocks/data/idp/idx/enroll-profile-new-custom-labels.json:25]: No, I created a ticket for it to be handled in Gen3, but this change is reflected here because I merged from master into this branch. Its just reflecting that this change didnt exist in the 0.5 branch.

**Comment by glenfannin-okta** [comment]
https://github.com/okta/okta-signin-widget/pull/3265#discussion_r1219986214

I created a ticket to do this work: https://oktainc.atlassian.net/browse/OKTA-617440
I also added the link you provided 
cc: @lesterchoi-okta

**Review by lesterchoi-okta — APPROVED** [review]
Looks good with the latest changes

**Comment by glenfannin-okta** [comment]
Created a new branch to clean up commit history:
https://github.com/okta/okta-signin-widget/pull/3275

Closing this PR in lieu of the above PR.
